src/main/java/org/jenkinsci/plugins/buildnamesetter/BuildNameSetter.java

```diff
@@ -26,48 +28,106 @@
 public class BuildNameSetter extends BuildWrapper implements MatrixAggregatable {
 
     public final String template;
-
+    public final String matrixTemplate;
+    private boolean matches;
+    private int scenario = -1;
+    private boolean started = false;
+    
     @DataBoundConstructor
+    public BuildNameSetter(String template, String matrixTemplate) {
+        this.template = template;
+        this.matrixTemplate = matrixTemplate;
+        matches = template.equals(matrixTemplate);
+        
+        if(matches || matrixTemplate.length()>0 && template.length()==0){
+        	scenario = 0;
+        }else if (matrixTemplate.length()>0 && template.length()>0){
+        	scenario = 1;
+        }else if (matrixTemplate.length()==0 && template.length()>0){
```

Comment:
The plugin's current coding style is to add spaces before open curly braces and after closing curly braces, so this should probably be made consistent.